packages/vitest/src/node/reporters/verbose.ts

```diff
@@ -9,45 +10,62 @@ export class VerboseReporter extends DefaultReporter {
   protected verbose = true
   renderSucceed = true
 
-  printTask(task: Task): void {
+  printTestModule(module: TestModule): void {
+    // still print the test module in watch mode,
+    // but don't print it in the CLI because we
+    // print all the tests when they finish
+    // instead of printing them when the test file finishes
     if (this.isTTY) {
-      return super.printTask(task)
+      return super.printTestModule(module)
     }
+  }
+
+  onTestCaseResult(test: TestCase): void {
+    super.onTestCaseResult(test)
+
+    // don't print tests in watch mode as they go, only print them
+    // in the CLI when they finish
+    if (this.isTTY) {
+      return
+    }
```

Comment:
```suggestion
    // don't print tests in TTY as they go, only print them
    // in the CLI when they finish
    if (this.isTTY) {
      return
    }
```

Reply:
Btw I think this difference is related to how Vitest v2 and older versions worked. Verbose reporter was constantly clearing and updating screen, so it needed different logic for non-TTY where clearing is not possible.

Now that v3 introduced the summary, we no longer update the reporter output at all. Summary is the only piece that is updated during run. We should probably make verbose reporter work 100% identical on non-TTY and TTY.

Reply:
We do print the file summary though which looks wierd in this case. Should we just not print when file starts/finish in verbose reporter?

Reply:
In short, verbose reporter prints tests as they finish, but the base reporter prints them when all of them are finished inside the test file

If we remove the `isTTY` check, then all tests will be printed and then the test file will also print them again

Reply:
Yep I'm not saying we should remove this TTY check here. Instead make verbose reporter identical in TTY and non-TTY. But that would be breaking change that we can intorduce in v4 if we want.

So currently:
- In non-TTY verbose reporter reports **test cases** immeadiatelly as they finish
- In TTY verbose reporter reports **test modules** as they finish

Base reporter shows test summary in both non-TTY and TTY. In TTY there's `summary` that's shown during test run. After test run it's cleared and we renderer the typical test summary that's been there since v0. It's the same summary in TTY and non-TTY.